sympy/solvers/solveset.py

```diff
@@ -865,15 +866,13 @@ def solveset(f, symbol=None):
     elif not symbol.is_Symbol:
         raise ValueError('A Symbol must be given, not type %s: %s' % (type(symbol), symbol))
 
-    real = (symbol.is_real is True)
-
     f = sympify(f)
 
     if f is S.false:
         return EmptySet()
 
     if f is S.true:
-        if real:
+        if domain is S.Reals:
             return S.Reals
         else:
             return S.Complexes
```

Comment:
If `f is S.true` we can simply return the `domain` i.e `return domain`, since may be the user may have defined the domain as: `Interval(1, 10)` (otherwise it would return `S.Complexes`). So i think this block of code can simply be replaced by `if f is S.true: return domain`.